standard/expressions.md

```diff
@@ -2256,6 +2339,8 @@ The binding-time processing of an indexer access of the form `P[A]`, where `P` i
 
 Depending on the context in which it is used, an indexer access causes invocation of either the get accessor or the set accessor of the indexer. If the indexer access is the target of an assignment, the set accessor is invoked to assign a new value ([§12.21.2](expressions.md#12212-simple-assignment)). In all other cases, the get accessor is invoked to obtain the current value ([§12.2.2](expressions.md#1222-values-of-expressions)).
 
+> *Note*: An implementation is required to provide an instance indexer member with a single parameter of type `Index` for any type that meets the criteria specified in §indexable-sequence-impl-support-for-index. An implementation is required to provide an instance indexer member with a single parameter of type `Range` for any type that meets the criteria specified in §indexable-sequence-impl-support-for-range. *end note*
```

Comment:
Because this is *required*, should it be normative? Should we state that the compiler must translate those expressions into the equivalent expressions using the integral types?

Reply:
This *Note* contradicts the clauses it references. With @jskeet as to the need to rewrite (normatively) here. Whether it describes “as if an indexer” exists (as in the referenced clauses), an argument conversion (as if) some call to `GetOffset()` is called (as in the changes currently suggested to array indexing), or as some kind of “target type conversion” (as [here](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/ranges#adding-index-and-range-support-to-existing-library-types)) is up for discussion. Hopefully we can describe array and non-array indexing behaviour very similarly.